src/core/utils/color/ColorValue.ts

```diff
@@ -0,0 +1,709 @@
+    scratchUInt32[0] = tbgr;
+    return {
+      b: scratchBytes[2],
+      g: scratchBytes[1],
+      r: scratchBytes[0],
+      t: scratchBytes[3],
+    };
+  }
+
+  /** The color value of this ColorValue as an integer in the form 0xTTBBGGRR (red in the low byte) */
+  public get tbgr(): number {
+    return this._tbgr;
+  }
+
+  /** Get the value of the color as a number in 0xAABBGGRR format (i.e. red is in low byte). Transparency (0==fully opaque) converted to alpha (0==fully transparent).  */
+  public getAbgr(): number {
+    return ColorValue.getAbgr(this._tbgr);
+  }
+
+  /** Get the value of a 0xTTBBGGRR color as a number in 0xAABBGGRR format (i.e. red is in low byte). Transparency (0==fully opaque) converted to alpha (0==fully transparent).  */
+  public static getAbgr(tbgr: number): number {
+    scratchUInt32[0] = tbgr;
+    scratchBytes[3] = 255 - scratchBytes[3];
+    return scratchUInt32[0];
+  }
+
+  /** Get the RGB value of the color as a number in 0xRRGGBB format (i.e blue is in the low byte). Transparency is ignored. Value will be from 0 to 2^24 */
+  public getRgb(): number {
+    return ColorValue.getRgb(this._tbgr);
+  }
+
+  /** Get the RGB value of the 0xTTBBGGRR color as a number in 0xRRGGBB format (i.e blue is in the low byte). Transparency is ignored. Value will be from 0 to 2^24 */
+  public static getRgb(tbgr: number): number {
+    scratchUInt32[0] = tbgr;
+    return (scratchBytes[0] << 16) + (scratchBytes[1] << 8) + scratchBytes[2];
+  }
+
+  /** Return a copy of this ColorValue with the specified alpha component.
+   * @param alpha the new alpha value as an integer between 0-255.
+   * @returns A ColorValue with equivalent red, green, and blue components to this one but with the specified alpha.
+   */
+  public withAlpha(alpha: number): ColorValue {
+    const tbgr = ColorValue.withAlpha(this._tbgr, alpha);
+    return tbgr === this._tbgr ? this : ColorValue.fromTbgr(tbgr);
+  }
+
+  /** Return a color equivalent to the specified 0xTTBBGGRR but with modified alpha component.
+   * @param alpha the new alpha value as an integer between 0-255.
+   * @returns The 0xTTBBGGRR value equivalent to `tbgr` but with the specified alpha.
+   */
+  public static withAlpha(tbgr: number, alpha: number): number {
+    scratchUInt32[0] = tbgr;
+    scratchBytes[3] = 255 - (alpha | 0);
+    return scratchUInt32[0];
+  }
+
+  /** Get the alpha value for this ColorValue. Will be between 0-255 */
+  public getAlpha(): number {
+    return ColorValue.getAlpha(this._tbgr);
+  }
+
+  /** Extract the alpha value from a 0xTTBBGGRR color. */
+  public static getAlpha(tbgr: number): number {
+    scratchUInt32[0] = tbgr;
+    return 255 - scratchBytes[3];
+  }
+
+  /** True if this ColorValue is fully opaque. */
+  public get isOpaque(): boolean {
+    return ColorValue.isOpaque(this._tbgr);
+  }
+
+  /** True if the specified 0xTTBBGGRR color is fully opaque. */
+  public static isOpaque(tbgr: number): boolean {
```

Comment:
Not sure this is needed.. Pretty simple and users can easily check that on their end. We provide ability to get alpha, so..